public/src/components/campaign/view_campaign.jsx

```diff
@@ -0,0 +1,101 @@
+import React, { Component } from 'react';
+import { Button, Row, Col } from 'react-bootstrap';
+import { Doughnut, Bar, HorizontalBar } from 'react-chartjs-2';
+import { outcomeDataFormat, statusDataFormat, responsesDataFormat, chartOptions, getQuestionNames } from '../../helpers/metrics_template';
+
+export default class ViewCampaign extends Component {
+
+
+  componentDidMount() {
+    if (this.props.admin_campaigns === undefined) {
+      const { id } = this.props.match.params;
+      this.props.fetchCampaign(id);
+      this.props.fetchCampaignMetrics(id);
+    }
+  }
+
+  render() {
+    const { current_campaign,
+            current_campaign_metrics,
+            current_script,
+            current_questions,
+            history } = this.props;
+    const { name, title, status, description, script_id } = current_campaign;
+    const { name: scriptName } = current_script;
+    let outcomeData;
+    let statusData;
+    let responsesData;
+    let questionNames;
+    if (current_campaign_metrics.metricsObj) {
+      outcomeData = outcomeDataFormat(current_campaign_metrics.metricsObj.outcome_distribution);
+      statusData = statusDataFormat(current_campaign_metrics.metricsObj.status_distribution);
+      responsesData = responsesDataFormat(current_campaign_metrics.metricsObj.response_stats);
+    }
+    if (current_questions && current_questions.length) {
+      questionNames = getQuestionNames(current_questions);
+    }
+    return (
+      <div>
```

Comment:
TODO: let's add to the todo for carry over that this should be broken out in to multiple components